sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc

```diff
@@ -674,69 +681,108 @@ _Complexity_: Let `N` be `last - first`. `O(N*log(N)*log(N))` comparisons.
 the default sorting algorithm defined by the sorter calling by `joint_sort`.
 `range_size` represents a range size for sorting,
 e.g. `last-first` from `operator()` arguments.
+It must be called within a SYCL kernel.
 Result depends on the `scope` parameter:
 use `sycl::memory_scope::work_group` to get memory size required
 for each work-group;
 use `sycl::memory_scope::sub_group` to get memory size required
 for each sub-group.
 If other `scope` values are passed, behavior is unspecified.
 
-(5) Implements a default sorting algorithm to be called by
+(4) Returns size of temporary memory (in bytes) that is required by
+the default sorting algorithm defined by the sorter calling by `joint_sort`
+depending on `d`.

```

Comment:
(3) contains the following phrase: `It must be called within a SYCL kernel.` So, (3) is supposed to be called within the SYCL kernel if someone wants to allocate static shared memory: we don't need a SYCL device parameter since it's automatically depends on the SYCL kernel. (4) must be called out of SYCL kernel if users want to create a global memory or local memory on host and reuse it in other kernels

Reply:
(3) is no longer `constexpr` so I can't see any way it could be used to allocate static shared memory.  How do you imagine this being used?

But you're right that I missed the difference in preconditions here.  If you end up keeping overload (3), I think you should split this out into separate _Returns_, _Preconditions_, and so on.  That would make it more obvious what the preconditions are.

Reply:
I see. Sorry, after 3 months I forgot original intention :) So, we can keep `constexpr` for (3) to allow users allocate static shared memory. However, I think it can overcomplicate things for users. Since this is an experimental feature, we can just remove (3) and keep only (4) and recommend users call `memory_required` on host only for `default_sorters`. For `radix_sorters` this method can be used within the kernel and on host both. What do you think?

Reply:
> If I understand correctly, we can explain this to users by saying that the default sorter query is a device-dependent runtime query because we don't know what the default sorting algorithm is

Right the exact algorithm depends on the device.

> different implementations of radix sort (for different devices) might also end up wanting different amounts of memory.

Potentially, in future, might be, but currently, there are no plans to make it dependent on the device.